src/runtime/serialization/serialization.cpp

```diff
@@ -0,0 +1,115 @@
+#include "hipSYCL/runtime/serialization/serialization.hpp"
+#include "hipSYCL/runtime/backend.hpp"
+#include "hipSYCL/runtime/dag.hpp"
+#include "hipSYCL/runtime/device_id.hpp"
+#include "hipSYCL/runtime/operations.hpp"
+#include "hipSYCL/sycl.hpp"
+#include "hipSYCL/sycl/access.hpp"
+#include "hipSYCL/sycl/item.hpp"
+#include "hipSYCL/sycl/range.hpp"
+#include "hipSYCL/sycl/serialization/serialization.hpp"
+
+#include <ostream>
+#include <map>
+
+namespace hipsycl::rt {
+
+std::ostream &operator<<(std::ostream &out, const hardware_platform value) {
+  static std::map<hardware_platform, std::string> strings;
+  if (strings.size() == 0) {
+    strings[hardware_platform::cpu] = "cpu";
+    strings[hardware_platform::cuda] = "cuda";
+    strings[hardware_platform::rocm] = "rocm";
+  }
+  return out << strings[value];
+}
+
+std::ostream &operator<<(std::ostream &out, const api_platform value) {
+  static std::map<api_platform, std::string> strings;
+  if (strings.size() == 0) {
+#define INSERT_ELEMENT(p) strings[p] = #p
+    INSERT_ELEMENT(api_platform::hip);
+    INSERT_ELEMENT(api_platform::openmp_cpu);
+#undef INSERT_ELEMENT
```

Comment:
I think we decided on not using that macro in favor of the solution above, right?

Reply:
I overlooked this one, thanks!